From bcf2a309f250c927e46d845f4b51a5819aaee14f Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 13 Jan 2022 15:36:39 +0200
Subject: [PATCH 06/25] mmc: fsl_esdhc_imx: Guard the access to scr register

SCR register is available when not using a reduced map of
eSDHC.

Issue: ALB-4149
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/mmc/fsl_esdhc_imx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/fsl_esdhc_imx.c b/drivers/mmc/fsl_esdhc_imx.c
index 03ec00dc5c..ac68ac87f7 100644
--- a/drivers/mmc/fsl_esdhc_imx.c
+++ b/drivers/mmc/fsl_esdhc_imx.c
@@ -1145,7 +1145,7 @@ static int esdhc_init_common(struct fsl_esdhc_priv *priv, struct mmc *mmc)
 	esdhc_write32(&regs->dllctrl, 0x0);
 #endif
 
-#ifndef ARCH_MXC
+#if !defined(ARCH_MXC) && !defined(SDHC_REDUCED_MAP)
 	/* Enable cache snooping */
 	esdhc_write32(&regs->scr, 0x00000040);
 #endif
-- 
2.17.1

